### ISSUE TITLE
Generate Networks for Analysis

### ISSUE THREAD

**Issue by ThomJ130** [sharifware/Large-Scale-Design-and-Analysis-of-Neural-Networks#1]
The system would produce $k$ neural networks, given the neural architecture and training data. The neural networks must be instantiated and trained. 

The neural networks must fit a success criteria. We will leave the implementation of the criteria as hard-coded but if you can make it user-defined that would be best (e.g. lambda expression).
The hard-coded success criteria would be a loss on the test-set must be less than some amount $\epsilon$. You could also parameterize the test-set's size as a % of the training-set, but it must at least be 10%.
A maximum number of iterations must also be specified to prevent wasting time on a bad solution.

Having an optional max-attempts may be a good idea. Since the success criteria may not be met for each network, we will often generate more than $k$ total networks (while throwing out the bad).
